lib/matrix.js

```diff
@@ -358,8 +704,32 @@ var init = function(exports){
             var params = {
                 limit: limit
             };
+            var self = this;
             return this._doAuthedRequest(
-                callback, "GET", "/initialSync", params
+                function(err, data) {
+                    if (self.store) {
+                        // intercept the results and put them into our store
+                        self.store.setPresenceEvents(data.presence.map(
+                            function(event) {
+                                return new MatrixEvent(event);
+                            }
+                        ));
+                        for (var i = 0 ; i < data.rooms.length; i++) {
+                            self.store.setStateEvents(data.rooms[i].state.map(
+                                function(event) {
+                                    return new MatrixEvent(event);
+                                }
+                            ));
+                            self.store.setEvents(data.rooms[i].messages.chunk.map(
```

Comment:
As much as I love `Array.map` it is in ECMAScript 5 meaning this won't work in IE8, which some certain.. corporate environments still use. Given we can get by without it (helper functions yay), we should probably keep compatibility.